frontend/src/__tests__/contexts/NotificationContext.test.tsx

```
@@ -0,0 +1,283 @@
+    return (
+        <div>
+            <span data-testid="unread-count">{context.unreadCount}</span>
+            <span data-testid="total-count">{context.notifications.length}</span>
+            <ul>
+                {context.notifications.map(n => (
+                    <li key={n.id} data-testid={`notification-${n.id}`}>
+                        {n.message}
+                    </li>
+                ))}
+            </ul>
+            <button onClick={() => context.markAsRead(1)}>Mark Read</button>
```

Biome (2.3.13)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)